lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -421,7 +468,32 @@ computeTxSize proxy action cs =
           where
             chaff = L8.pack (show ix) <> BL.fromStrict txid
 
-    wits = SL.WitnessSet (Set.union addrWits certWits) mempty mempty
+    byronWits = Set.map dummyWitnessUniq $ Set.fromList (fst <$> CS.inputs cs)
+      where
+        dummyWitness :: BL.ByteString -> SL.BootstrapWitness TPraosStandardCrypto
+        dummyWitness chaff =
+            SL.makeBootstrapWitness credential signingKey addrAttr
+          where
+            credential = Hash.UnsafeHash $ L8.toStrict chaff
+            addrAttr = Byron.mkAttributes $ Byron.AddrAttributes
+                       (Just $ Byron.HDAddressPayload (bloatChaff keyLen))
```

Comment:
Why `keyLen` here :thinking:  ? The HDAddressPayload is a ChaChaPoly digest + nonce that totals 28 bytes + 2 bytes because it's CBOR-encoded. I believe the `keyLen` is 32 bytes here so it's probably not a bit deal, but that is quite misleading and error-prone.

More importantly, there's only a HDPayload for /Byron/ type of addresses, Icarus do not.

Reply:
addressed here https://github.com/input-output-hk/cardano-wallet/pull/1849/commits/1209bca45f0c0dd9cd8615355eee19df11893e46